drivers/spi/spi_ll_stm32.c

```diff
@@ -98,13 +97,19 @@ static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 	if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
 		rx_frame = LL_SPI_ReceiveData8(spi);
 		if (spi_context_rx_on(&data->ctx)) {
-			UNALIGNED_PUT(rx_frame, (u8_t *)data->ctx.rx_buf);
+			if (data->ctx.rx_buf) {
+				UNALIGNED_PUT(rx_frame,
+					      (u8_t *)data->ctx.rx_buf);
+			}
 			spi_context_update_rx(&data->ctx, 1, 1);
```

Comment:
Why does this remain outside the new if statement, instead of moving inside? Similar questions below.

Reply:
I thought it should but it broke spi_loopback test. If the spi_context_update_rx() is not called, it will never move to the second rx_buf.

Reply:
I propose to move `spi_context_update_rx()` inside if and push another commit which will substitute null rx_bufs with `dummy_rx_bufs` in spi_loopback test. How does that sound to you?

Reply:
Interesting, so the .buf = NULL inside of the rx_bufs definitions (like in spi_rx_half_end()) in the spi_loopback test is breaking things, you mean?

If so, perhaps this is a bug in the SPI core? Haven't had time to look into it in detail, just to be clear.

Reply:
> Interesting, so the .buf = NULL inside of the rx_bufs definitions (like in spi_rx_half_end()) in the spi_loopback test is breaking things, you mean?

Yes, it causes bus error on some SoCs (e.g. stm32f103, stm32f334).

Reply:
Thank you for your response!

@tbursztyka, could the test possibly be structuring its buffers incorrectly from what you meant?

If not, perhaps this is a bug in the STM32 area.